File: src/QueReal.PL/Models/Quest/QuestGetResponse.cs
namespace QueReal.PL.Models.Quest
{
    public class QuestGetResponse
    {
        public Guid Id { get; set; }
        public string Title { get; set; }

        public DateTime CreateTime { get; set; }

        public DateTime UpdateTime { get; set; }

        public DateTime? ApprovedTime { get; set; }

        public List<QuestItemGetResponse> QuestItems { get; set; }
    }
}

File: frontend/src/app/common/api/models/quest/quest-get-response.ts
import { QuestItemGetResponse } from "./quest-item-get-response";

export class QuestGetResponse {
    public id: string = null!;

    public title: string = null!;

    public createTime: Date = null!;
    
    public updateTime: Date = null!;
    
    public approvedTime: Date | null = null;

    public questItems: QuestItemGetResponse[] = null!;
}
